### PR TITLE
Implement and Integrate `java-slang` CSEC Visualizer

### PR DESCRIPTION
**Pull request by xyliew25**
P.s., This PR to be ready for review after #2925 merged as there will be foreseeable merge conflicts in `JavaHelper.ts`. 

### Description

Implement and integrate the concomitant CSEC Visualizer of the [CSEC Machine from `java-slang`.](https://github.com/source-academy/java-slang/tree/main/src/ec-evaluator)

image

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Code quality improvements

### How to test

1. Select the Java language.
2. Input syntactically valid and well-typed Java programs into the code editor - a simple one like the one below or more interesting ones from [`java-slang/src/ec-evaluator/__tests__`](https://github.com/source-academy/java-slang/tree/main/src/ec-evaluator/__tests__)

    ```
    class Test {
        public static void main(String[] args) {
            int x = 1;
        }
    }
    ```
3. Head over to the CSE Machine tab.
4. Click on the Run button.

### Checklist

- [X] I have tested this code (I've only done manual inspections)
- [ ] I have updated the documentation (I've updated the intro text in the CSE Machine tab though the link to Java CSEC sublanguage specification is invalid)

### CODE REVIEW

File: src/commons/sideContent/content/SideContentCseMachine.tsx [hunk 1]
```diff
@@ -371,20 +429,28 @@ class SideContentCseMachineBase extends React.Component<CseMachineProps, State>
...
         </ButtonGroup>
       </HotKeys>
     );
   }
 
+  private zoomStage = (isZoomIn: boolean, multiplier: number) => {
+    if (this.isJava()) {
+      JavaCseMachine.zoomStage(isZoomIn, multiplier);
+    } else {
+      Layout.zoomStage(false, 5);
```
Comment (RichDom2185) [inline src/commons/sideContent/content/SideContentCseMachine.tsx:450]: Is this a typo?

```suggestion
      Layout.zoomStage(isZoomIn, multiplier);
```

File: src/commons/utils/JavaHelper.ts [hunk 2]
```diff
@@ -148,3 +157,43 @@ export async function javaRun(javaCode: string, context: Context) {
...
+    severity: ErrorSeverity.ERROR,
+    location: {
+      start: {
+        line: 0,
+        column: 0
+      },
+      end: {
+        line: 0,
+        column: 0
+      }
```
Comment (RichDom2185) [inline src/commons/utils/JavaHelper.ts:181]: Is this not planned to be supported?

**Review by RichDom2185 — COMMENTED** [review]
Hi, looking forward to seeing the CSEC machine for Java! I just have these two comments, thanks a lot!

File: src/commons/utils/JavaHelper.ts (hunk 2 above)
Comment (xyliew25) [reply src/commons/utils/JavaHelper.ts:181]: At least not for this PR, errors thrown by ECE in `java-slang` currently do not have source node information.

Comment (RichDom2185) [reply src/commons/utils/JavaHelper.ts:181]: Noted, thanks! Perhaps we can leave a TODO comment

Comment (xyliew25) [reply src/commons/utils/JavaHelper.ts:181]: Sure

**Review by RichDom2185 — APPROVED** [review]
LGTM, thanks a lot!

**Pull request merged**